seed-job/index_reader.py

```diff
@@ -0,0 +1,173 @@
+            if dependson is not None:
+                if isinstance(dependson, list):
+                    dependson_job = ','.join(dependson)
+                else:
+                    dependson_job = str(dependson)
+                dependson_job = dependson_job.replace(
+                    ':', '-').replace('/', '-')
+
+            if dependson_job == '':
+                dependson_job = 'none'
+        """
+
+    def process_desired_tag(self, desired_tag=None):
+        """
+        Process desired_tag for given project
+        """
+        if not desired_tag:
+            return "latest"
+        return desired_tag
+
+    def process_build_context(self, build_context=None):
+        """
+        Process build_context for given project
+        """
+        if not build_context:
+            return "./"
+        return build_context
+
+    def process_prebuild_script(self, prebuild_script=None):
+        """
+        Process prebuild_script for given project
+        """
+        if not prebuild_script:
+            return None
+        return prebuild_script
+
+    def process_pre_build_context(self, prebuild_context=None):
+        """
+        Process prebuild_context for given project
+        """
+        if not prebuild_context:
+            return None
+        return prebuild_context
+
+    def load_project_entry(self, entry):
+        """
+        Loads a container index entry in class objects
+        """
+        try:
+            self.appid = self.replace_dot_slash_colon_(entry['app-id'])
+            self.jobid = self.replace_dot_slash_colon_(entry['job-id'])
+
+            self.giturl = entry['git-url']
+            self.gitpath = entry['git-path']
+            self.gitbranch = entry['git-branch']
+            self.targetfile = entry['target-file']
+            self.dependson = self.process_depends_on(entry['depends-on'])
+            self.notifyemail = entry['notify-email']
+            self.desiredtag = self.process_desired_tag(entry["desired-tag"])
+            self.build_context = self.process_build_context(
+                entry["build-context"])
+            self.pre_build_context = self.process_pre_build_context(
+                entry["prebuild-context"])
+            self.pre_build_script = self.process_pre_build_script(
+                entry["prebuild-script"])
+        except Exception as e:
+            print ("Error processing container index entry.")
+            raise(e)
+
+    def pipeline_name(self):
+        """
+        Returns the pipeline name based on the project object values
+        """
+        return "{}-{}-{}".format(self.appid, self.jobid, self.desiredtag)
+
+
+class IndexReader(object):
+    """
+    Class for reading container index and utilities
+    """
+
+    def __init__(self, index, namespace):
+        """
+        Initialize class variable with index location
+        """
+        self.index = index
+        self.namespace = namespace
+
+    def read_yaml(self, filepath):
```

Comment:
This should go into a lib, as it can be used by multiple other functions, but ok for now